drivers/gpu/drm/i2c/adv7511.c

```diff
@@ -581,6 +591,29 @@ static int adv7511_irq_process(struct adv7511 *adv7511)
 			wake_up_all(&adv7511->wq);
 	}
 
+	/*
+	 * reset dsi TX/RX when transfer error
+	 */
+	ret = regmap_read(adv7511->regmap_cec, 0x48, &state);
+	if (ret < 0)
+		return ret;
+	if (powered && (!irq1 || state & BIT(2))) {
```

Comment:
I spent some more time looking at this patch today. And noticed an odd thing.

The reason why this conditional is true, is usually because we're powered on and the irq1 register is zero.
Is that intended? In all my testing so far, the reset seems to happen here only when an irq lands after we've powered on (regardless of why it triggered). In most cases, I'm seeing: irq0=0x4, or irq=0x2 and irq1=0x0 and state=0x0.

So from my reading of this, basically we're seeing the irqs becasue of the  "HDCP Authenticated" and  "EDID Ready Interrupt" bits. This doesn't align with the commented case were we're trying to reset due to a transfer error.

So I'm confused by this. Can you explain?